source/euler/hyperbolic_system.h

```diff
@@ -1313,5 +1339,159 @@ namespace ryujin
+
+       scalar_type density, pressure;
+       std::vector<scalar_type> momentum(dim);
+
+   //    std::cout << "inside calc before density partition" << std::endl;
+       //initialize partitions
+       density.reinit(offline_data_.scalar_partitioner(), mpi_communicator_);
+       pressure.reinit(offline_data_.scalar_partitioner(), mpi_communicator_);
+       for(unsigned int c=0; c < dim; c++)
+         momentum.at(c).reinit(offline_data_.scalar_partitioner(), mpi_communicator_);
+
+       dealii::Tensor<1,dim> normal_vector;
+       dealii::SymmetricTensor<2,dim> fluid_stress;
+       dealii::SymmetricTensor<2,dim> fluid_pressure;
+       dealii::Tensor<1,dim> forces;
+
+       dealii::FEFaceValues<dim> fe_face_values(fe, face_quadrature_formula,
+           dealii::update_values | dealii::update_quadrature_points | dealii::update_gradients |
+           dealii::update_JxW_values | dealii::update_normal_vectors); //the face values
+
+       double drag = 0.;
+       double lift = 0.;
+
+
+       // Create vectors that store the locally owned parts on every process
+       U.extract_component(density, 0);//extract density
+       U.extract_component(pressure, dim+1);//extract density
+
+       //extract momentum, and convert to velocity
+       for(int c=0; c<dim; c++)
+       {
+         int comp = c+1;//momentum is stored in positions [1,...,dim], so add one to c
+         U.extract_component(momentum.at(c), comp);
+       }
+
+       //extract energy
+       U.extract_component(pressure, dim+1);
+
+       //convert E to pressure
+       for (unsigned int k=0; k<offline_data_.n_locally_owned(); k++)
+       {
+
+         //calculate momentum norm squared
+         const double &E = pressure.local_element(k);
+         const double &rho = density.local_element(k);
+         double m_square = 0;
+         for(int d=0; d<dim; d++)
+           m_square += std::pow(momentum.at(d).local_element(k),2);
+
+         //pressure = (gamma-1)*internal_energy
+         pressure.local_element(k) =  (gamma() - 1.0) * (E - 0.5*m_square/rho);
+       }
+
+
+       if((pressure.has_ghost_elements() != true) || (density.has_ghost_elements() != true)
+           ||(momentum.at(0).has_ghost_elements() != true) || (momentum.at(1).has_ghost_elements() != true))
+       {
+         //TODO: make this some type of exception
+         std::cout << "ghost elements not allowed " << std::endl;
+         std::cout << "pressure " << (pressure.has_ghost_elements() != true) << std::endl;
+         std::cout << "momx " << (momentum.at(0).has_ghost_elements() != true) << std::endl;
+         std::cout << "momy " << (momentum.at(1).has_ghost_elements() != true) << std::endl;
+         std::cout << "density " << (density.has_ghost_elements() != true) << std::endl;
+
+         exit(EXIT_FAILURE);//TODO: remove, for now, I want this to just quit the program
+       }
+       density.update_ghost_values();
+       pressure.update_ghost_values();
+       for(auto mom: momentum)
```

Comment:
Make it clear that you want to update the elements, not copies of them.
```suggestion
       for(auto &mom: momentum)
```